commit 8d49193a53b01dd058d1e64ef2768123ee233c97 (refs/changes/72/3072/2)
Author: Rui Loura <rui@joyent.com>
Date:   2017-12-18T23:12:07+00:00 (1 year, 10 months ago)
    
    AGENT-1086 Want to support *-rackN nic_tag suffixes in config-agent

diff --git a/CHANGES.md b/CHANGES.md
index 0d444c3..15d36b1 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,5 +1,11 @@
 # sdc-config-agent changelog
 
+## 1.6.0
+
+- [AGENT-1086] Support network per rack Manta deployments by allowing 
+  nic_tags with \<tag>-rackNN suffixes to override nic_tags with the same \<tag>
+  (e.g "manta-rack99" overrides "manta").
+
 ## 1.5.0
 
 - [TOOLS-1084] Support sync manifest update through SIGHUP using
diff --git a/Makefile b/Makefile
index 0b7e56c..841fa00 100644
--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,7 @@ JS_FILES	:= $(shell ls *.js) $(shell find cmd lib -name '*.js')
 JSL_CONF_NODE	 = tools/jsl.node.conf
 JSL_FILES_NODE   = $(JS_FILES)
 JSSTYLE_FILES	 = $(JS_FILES)
-JSSTYLE_FLAGS    = -o doxygen
+JSSTYLE_FLAGS    = -t 4 -o doxygen
 SMF_MANIFESTS_IN = smf/manifests/config-agent.xml.in
 
 NODE_PREBUILT_VERSION=v0.10.26
diff --git a/agent.js b/agent.js
index c45927f..91e11b3 100644
--- a/agent.js
+++ b/agent.js
@@ -152,6 +152,16 @@ async.waterfall([
 				if (nic.nic_tag) {
 					var NIC_TAG = nic.nic_tag.toUpperCase();
 					autoMetadata[NIC_TAG + '_IP'] = nic.ip;
+
+					/*
+					 * If there is a nic tag of the form <name>-RACK<number>,
+					 * it will override any similarly named nic tag (e.g.
+					 * "MANTA-RACK##" will override "MANTA" nic tags).
+					 */
+					if (NIC_TAG.search(/^[A-Z]+-RACK\d+$/) === 0) {
+						NIC_TAG = NIC_TAG.split('-')[0];
+						autoMetadata[NIC_TAG + '_IP'] = nic.ip;
+					}
 				}
 			}
 
